sympy/physics/control/tests/test_lti.py

```diff
@@ -0,0 +1,473 @@
+    assert G5/2 == TransferFunction(s + 6, 2*s - 10, s)
+    assert G5/(s**2) == TransferFunction(s + 6, s**2*(s - 5), s)
+    assert (s - 4*s**2)/G2 == TransferFunction((s - 5)*(-4*s**2 + s), s + 1, s)
+    assert 0/G4 == TransferFunction(0, p + 4, p)
+
+    raises(ValueError, lambda: G3 / Matrix([1, 2, 3]))
+    raises(ValueError, lambda: G6 / 0)
+    raises(ValueError, lambda: G3 / G5)
+
+    p1 = (9 - s**3)*(s - 5)*(s + 1)*(s + 6) + (s - 5)*(s + 1)*(s + 3)**2
+    p2 = (9 - s**3)*(s - 5)**2*(s + 3)
+    assert G1*G2 + G5/G6 == TransferFunction(p1, p2, s)
+
+    G3_ = TransferFunction(s**4 - 2*s**2 + 1, s - 1, s)
+    p3 = (s - 5)*(s - 1)*(s + 3)
+    p4 = (9 - s**3)*(s + 1)*(s**4 - 2*s**2 + 1)
+    assert G1/G2/G3_ == TransferFunction(p3, p4, s)
+
+    p5 = (s - 5)*(s - 1)*(s + 3)**2 - (9 - s**3)*(s + 1)**2*(s**4 - 2*s**2 + 1)
+    p6 = (9 - s**3)*(s - 1)*(s + 1)*(s + 3)
+    assert G1/G2 - G3_/G6 == TransferFunction(p5, p6, s)
+
+    omega_o, zeta = symbols('omega_o, zeta')
+    G7 = TransferFunction(a0*s**p + a1*p**s, a2*p*zeta - s, zeta)
+    G8 = TransferFunction(omega_o**2, s**2 + p*omega_o*zeta*s + omega_o**2, zeta)
+    G9 = TransferFunction(zeta - s**3, zeta + p**4, zeta)
+    num = (p**4 + zeta)*(a0*s**p + a1*p**s)*(omega_o**2 + omega_o*p*s*zeta + s**2)
+    den = (omega_o**2*(p**4 + zeta) - (-s**3 + zeta)*(omega_o**2 + omega_o*p*s*zeta + s**2))
+    assert G7 / (G8 - G9) == TransferFunction(num, den*(a2*p*zeta - s), zeta)
+
+
+def test_TransferFunction_is_proper():
+    omega_o, zeta, tau = symbols('omega_o, zeta, tau')
+    G1 = TransferFunction(omega_o**2, s**2 + p*omega_o*zeta*s + omega_o**2, omega_o)
+    G2 = TransferFunction(tau - s**3, tau + p**4, tau)
+    G3 = TransferFunction(a*b*s**3 + s**2 - a*p + s, b - s*p**2, p)
+    G4 = TransferFunction(b*s**2 + p**2 - a*p + s, b - p**2, s)
+    assert G1.is_proper
+    assert G2.is_proper
+    assert G3.is_proper
+    assert not G4.is_proper
+
+
+def test_TransferFunction_is_strictly_proper():
+    omega_o, zeta, tau = symbols('omega_o, zeta, tau')
+    tf1 = TransferFunction(omega_o**2, s**2 + p*omega_o*zeta*s + omega_o**2, omega_o)
+    tf2 = TransferFunction(tau - s**3, tau + p**4, tau)
+    tf3 = TransferFunction(a*b*s**3 + s**2 - a*p + s, b - s*p**2, p)
+    tf4 = TransferFunction(b*s**2 + p**2 - a*p + s, b - p**2, s)
+    assert not tf1.is_strictly_proper
+    assert not tf2.is_strictly_proper
+    assert tf3.is_strictly_proper
+    assert not tf4.is_strictly_proper
+
+
+def test_TransferFunction_is_biproper():
+    tau, omega_o, zeta = symbols('tau, omega_o, zeta')
+    tf1 = TransferFunction(omega_o**2, s**2 + p*omega_o*zeta*s + omega_o**2, omega_o)
+    tf2 = TransferFunction(tau - s**3, tau + p**4, tau)
+    tf3 = TransferFunction(a*b*s**3 + s**2 - a*p + s, b - s*p**2, p)
+    tf4 = TransferFunction(b*s**2 + p**2 - a*p + s, b - p**2, s)
+    assert tf1.is_biproper
+    assert tf2.is_biproper
+    assert not tf3.is_biproper
+    assert not tf4.is_biproper
```

Comment:
`Series` and `Parallel` should also have the three "proper" methods.

Reply:
Sure. Another method that could be added is `transfer_function`, which gives the resultant transfer function from a Series or Parallel object.
```python
>>> Series(TF1, TF2).doit() == Series(TF1, TF2).transfer_function
```

Reply:
I think that doit() is sufficient. There is no need for a redudant method.